airsonic-main/src/main/java/org/airsonic/player/service/search/SearchServiceTermination.java

```diff
@@ -0,0 +1,199 @@
+/*
+ This file is part of Airsonic.
+
+ Airsonic is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ Airsonic is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with Airsonic.  If not, see <http://www.gnu.org/licenses/>.
+
+ Copyright 2016 (C) Airsonic Authors
+ Based upon Subsonic, Copyright 2009 (C) Sindre Mehus
+ */
+
+package org.airsonic.player.service.search;
+
+import org.airsonic.player.dao.AlbumDao;
+import org.airsonic.player.dao.ArtistDao;
+import org.airsonic.player.domain.Album;
+import org.airsonic.player.domain.Artist;
+import org.airsonic.player.domain.MediaFile;
+import org.airsonic.player.domain.ParamSearchResult;
+import org.airsonic.player.domain.SearchResult;
+import org.airsonic.player.service.MediaFileService;
+import org.airsonic.player.service.SettingsService;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.index.Term;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import static org.springframework.util.ObjectUtils.isEmpty;
+
+/**
+ * Termination used by SearchService.
+ * 
+ * Since SearchService operates as a proxy for storage (DB) using lucene,
+ * there are many redundant descriptions different from essential data processing.
+ * This class is a transfer class for saving those redundant descriptions.
+ * 
+ * Exception handling is not termination,
+ * so do not include exception handling in this class.
+ */
+@Component
+public class SearchServiceTermination {

```

Comment:
This definitely needs a different name. It seems this class is mainly a holder for various utility functions related to searching. A generic name therefore might be SearchServiceUtilities?

Reply:
I revised that according to your advice.